generics/middleware/authenticator.js

```diff
@@ -6,166 +6,168 @@
-    rspObj.errMsg = CONSTANTS.apiResponses.TOKEN_MISSING_MESSAGE;
-    rspObj.responseCode = HTTP_STATUS_CODE["unauthorized"].status;
-    return res.status(HTTP_STATUS_CODE["unauthorized"].status).send(respUtil(rspObj));
-  }
-
-  rspObj.errCode = CONSTANTS.apiResponses.TOKEN_INVALID_CODE;
-  rspObj.errMsg = CONSTANTS.apiResponses.TOKEN_INVALID_MESSAGE;
-  rspObj.responseCode = HTTP_STATUS_CODE["unauthorized"].status;
-
-  // <---- For Elevate user service user compactibility ---->
-  try {
-    decodedToken = jwt.verify(token, process.env.ACCESS_TOKEN_SECRET)
-  } catch (err) {
-    return res.status(HTTP_STATUS_CODE["unauthorized"].status).send(respUtil(rspObj));
-  }
-  if (!decodedToken) {
-    return res.status(HTTP_STATUS_CODE["unauthorized"].status).send(respUtil(rspObj));
-  }
-
-  req.userDetails = {
-    userToken : token,
-    userInformation : {
-      userId : decodedToken.data.id.toString(),
-      userName : decodedToken.data.name,
-      // email : decodedToken.data.email, //email is removed from token
-      firstName : decodedToken.data.name
-    }
-  };
-
-  console.log(req.userDetails,'req.userDetails')
-  next();
-  
-  // var decoded = jwt.decode(token, { complete: true });
-  // if(decoded === null || decoded.header === undefined){
-  //   return res.status(HTTP_STATUS_CODE["unauthorized"].status).send(respUtil(rspObj));
-  // }
-
-  // const kid = decoded.header.kid
-  // let cert = "";
-  // let path = keyCloakPublicKeyPath + kid;
-    
-  // if (fs.existsSync(path)) {
-
-  //   let accessKeyFile  = await fs.readFileSync(path);
-
-  //   if(accessKeyFile) {
-  //     if(!accessKeyFile.includes(PEM_FILE_BEGIN_STRING)){
-  //       cert = PEM_FILE_BEGIN_STRING+"\n"+accessKeyFile+"\n"+PEM_FILE_END_STRING;
-  //     }else {
-  //       cert = fs.readFileSync(path);
-  //     }  
-  //   }
-    
-  //   jwt.verify(token, cert, { algorithm: ['sha1', 'RS256', 'HS256'] }, function (err, decode) {
-  //     if (err) {
-  //       return res.status(HTTP_STATUS_CODE["unauthorized"].status).send(respUtil(rspObj));
-  //     }
-
-  //     if (decode !== undefined) {
-  //       const expiry = decode.exp;
-  //       const now = new Date();
-  //       if (now.getTime() > expiry * 1000) {
-  //         return res.status(HTTP_STATUS_CODE["unauthorized"].status).send(respUtil(rspObj));
-  //       }
-
-  //       req.userDetails = {
-  //         userToken : token,
-  //         userInformation : {
-  //           userId : decode.sub.split(":").pop(),
-  //           userName : decode.preferred_username,
-  //           email : decode.email,
-  //           firstName : decode.name
-  //         }
-  //       };
-
-  //       next();
-  //     } else {
-  //       return res.status(HTTP_STATUS_CODE["unauthorized"].status).send(respUtil(rspObj));
-  //     }
-  //   });
-  // } else {
-  //   return res.status(HTTP_STATUS_CODE["unauthorized"].status).send(respUtil(rspObj));
-  // }
-};
\ No newline at end of file
+	'host',
```

Comment:
@Prajwal17Tunerlabs Is there any code change in this file?

Reply:
yes , below is the code change 
`	req.userDetails = {
		userToken: token,
		userInformation: {
			userId: decodedToken.data.id.toString(),
			userName: decodedToken.data.name,
			**organizationId: decodedToken.data.organization_id,**
			// email : decodedToken.data.email, //email is removed from token
			firstName: decodedToken.data.name,
		},
	}`
have added organizationId to userDetails